### ISSUE TITLE
[Question]test on speakers untrained

### ISSUE THREAD

**Issue by Aurora11111** [google/uis-rnn#40]
## Describe the question
when I tested my own datasets which  haven't been trained ,the result is really bad,can you give me some suggestion?

## My background

Have I read the `README.md` file?
* yes

Have I searched for similar questions from closed issues?
* yes

Have I tried to find the answers in the paper *Fully Supervised Speaker Diarization*?
* yes

Have I tried to find the answers in the reference *Speaker Diarization with LSTM*?
* yes

Have I tried to find the answers in the reference *Generalized End-to-End Loss for Speaker Verification*?
* yes

**Comment by MuruganR96** [comment]
Pytorch TIMIT d-vector embeddings for my own datasets not giving good results for UIS-RNN archietecture. then how can we improvise accuracy? any suggestions for me also.?

@Aurora11111  sir, i also facing this issue. :+1:

**Comment by wq2012** [comment]
For training the d-vector model, I would suggest using all possible speaker-labelled datasets you could find, including:

- VCTK
- LibriSpeech
- VoxCeleb
- VoxCeleb2
- Datasets with similar acoustic environments to your own

The first 4 sum up to almost 10K different speakers. Although it's still much smaller than what we use internally (100K+ speakers), it would produce reasonably good results. (according to our experience, training d-vector model with less than 3~5K speakers is usually bad)

Then for training UIS-RNN, you could consider using:

- NIST SRE disk-6 + disk-8
- ICSI
- Fisher
- Datasets with similar acoustic environment and dialogue style to your own

The last one should not cover the same speakers to your testing test, but should have the same acoustic environment and dialogue style to your testing set. UIS-RNN is for supervised diarization, meaning you cannot expect it to work when trained in one domain and tested in a totally different domain:
https://www.youtube.com/watch?v=pGkqwRPzx9U&t=7m56s

**Comment by Aurora11111** [comment]
@wq2012 thanks for your suggestions,I will have a try.
